Build fails on rust 1.91

# Summary

This crate builds in 1.90 but no more in 1.91 (also not in 1.93.0-nightly). There seems to be some kind of regression with lifetime checking.

(It would be nice if someone would verify this, so it's not a my_computer thing)


# The error

```commandline
error[E0310]: the parameter type `S` may not live long enough
   --> src/lib.rs:664:5
    |
664 | /     {
665 | |         let search_stream = self
666 | |             .ldap
667 | |             .streaming_search(base, scope, filter.filter().as_str(), attributes)
...   |
676 | |         to_native_stream(search_stream)
677 | |     }
    | |     ^
    | |     |
    | |_____the parameter type `S` must be valid for the static lifetime...
    |       ...so that the type `S` will meet its required lifetime bounds
    |
help: consider adding an explicit lifetime bound
    |
663 |         S: AsRef<str> + Send + Sync + 'a + 'static,
    |                                          +++++++++

error[E0310]: the parameter type `A` may not live long enough
   --> src/lib.rs:664:5
    |
664 | /     {
665 | |         let search_stream = self
666 | |             .ldap
667 | |             .streaming_search(base, scope, filter.filter().as_str(), attributes)
...   |
676 | |         to_native_stream(search_stream)
677 | |     }
    | |     ^
    | |     |
    | |_____the parameter type `A` must be valid for the static lifetime...
    |       ...so that the type `A` will meet its required lifetime bounds
    |
help: consider adding an explicit lifetime bound
    |
662 |         A: AsRef<[S]> + Send + Sync + 'a + 'static,
    |                                          +++++++++
```


# Possible culprits

- https://github.com/rust-lang/rust/issues/148854
- https://github.com/rust-lang/rust/issues/147285
  - This was already in 1.90 though


# Workaround

Use rust 1.90 for now.

Perhaps something could be done to the code as well but surely the suggested fix of adding a `'static` bound is wrong?

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -656,7 +656,7 @@ impl LdapClient {
         scope: Scope,
         filter: &F,
         attributes: A,
-    ) -> Result<impl Stream<Item = Result<Record, Error>> + use<'a, F, A, S>, Error>
+    ) -> Result<impl Stream<Item = Result<Record, Error>> + 'a + use<'a, F, A, S>, Error>
     where
         F: Filter,
         A: AsRef<[S]> + Send + Sync + 'a,
@@ -1789,7 +1789,7 @@ where
 /// A helper to create native rust streams out of `ldap3::SearchStream`s.
 fn to_native_stream<'a, S, A>(
     ldap3_stream: SearchStream<'a, S, A>,
-) -> Result<impl Stream<Item = Result<Record, Error>> + use<'a, S, A>, Error>
+) -> Result<impl Stream<Item = Result<Record, Error>> + 'a + use<'a, S, A>, Error>
 where
     S: AsRef<str> + Send + Sync + 'a,
     A: AsRef<[S]> + Send + Sync + 'a,

